src/app/clusters/air-pressure-measurement-server/air-pressure-measurement-server.cpp

```diff
@@ -0,0 +1,84 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "air-pressure-measurement-server.h"
+
+#include <app/util/af.h>
+
+#include "gen/att-storage.h"
+#include "gen/attribute-id.h"
+#include "gen/attribute-type.h"
+#include "gen/cluster-id.h"
+#include "gen/command-id.h"
+
+#include <support/logging/CHIPLogging.h>
+
+#ifndef emberAfAirPressureMeasurementClusterPrintln
+#define emberAfAirPressureMeasurementClusterPrintln(...) ChipLogProgress(Zcl, __VA_ARGS__);
+#endif
+
+EmberAfStatus emberAfAirPressureMeasurementClusterGetMeasuredValue(chip::EndpointId endpoint, uint16_t * measuredValue)
+{
+    return emberAfReadServerAttribute(endpoint, ZCL_AIR_PRESSURE_MEASUREMENT_CLUSTER_ID,
+                                      ZCL_AIR_PRESSURE_MEASUREMENT_MEASURED_VALUE_ATTRIBUTE_ID, (uint8_t *) measuredValue,
+                                      sizeof(*measuredValue));
+}
+
+static EmberAfStatus emberAfAirPressureMeasurementClusterSetAltitudeCallback(chip::EndpointId endpoint, int16_t altitude)
+{
+    return emberAfWriteServerAttribute(endpoint, ZCL_AIR_PRESSURE_MEASUREMENT_CLUSTER_ID,
+                                       ZCL_AIR_PRESSURE_MEASUREMENT_ALTITUDE_ATTRIBUTE_ID, (uint8_t *) &altitude,
+                                       ZCL_INT16S_ATTRIBUTE_TYPE);
+}
+
+void emberAfAirPressureMeasurementClusterServerInitCallback(chip::EndpointId endpoint)
+{
+    uint16_t measuredValue = 0;
+    EmberAfStatus status   = emberAfAirPressureMeasurementClusterGetMeasuredValue(endpoint, &measuredValue);
+    if (EMBER_ZCL_STATUS_SUCCESS != status)
+    {
+        status = emberAfAirPressureMeasurementClusterSetMeasuredValueCallback(endpoint, 10132);
+        if (EMBER_ZCL_STATUS_SUCCESS != status)
+        {
+            emberAfAirPressureMeasurementClusterPrintln("ERR: writing present value %x", status);
+        }
+    }
+
+    int16_t altitude = 0;
+    status           = emberAfAirPressureMeasurementClusterGetAltitudeValue(endpoint, &altitude);
+    if (EMBER_ZCL_STATUS_SUCCESS != status)
+    {
+        status = emberAfAirPressureMeasurementClusterSetAltitudeCallback(endpoint, 0);
```

Comment:
Does `0` has any special meaning here ? Or similarly to other sensors we should have a value that means:  `unknown` ?

Reply:
Yes 0 means that the accessory thinks that it is at sea level and will not correct the air pressure according to height above / below sea level